scripts/check_compliance.py

```diff
@@ -87,34 +87,21 @@ class MyCase(TestCase):
 
 class ComplianceTest:
     """
-    Main Test class
+    Base class for tests. Inheriting classes should have a run() method and set
+    these class variables:
 
-    """
-
-    _name = ""
-    _title = ""
-    _doc = "https://docs.zephyrproject.org/latest/contribute/"
+    name:
+      Test name
 
-    def __init__(self, suite, commit_range):
-        self.case = None
-        self.suite = suite
-        self.commit_range = commit_range
-        # get() defaults to None if not present
+    doc:
+      Link to documentation related to what's being tested
 
-    def prepare(self, where):
-        """
-        Prepare test case
-        :return:
-        """
-        self.case = MyCase(self._name)
+    path:
+      The path the test runs in. This is just informative and used in messages.
```

Comment:
`path` is vague; `tested_path`? (NOT the current directory as this comment leaves somewhat open for misinterpretation)

Reply:
Is `path_hint` fine? Changed it to that, and tweaked the documentation a bit.